resources/aws/elb.go

```diff
@@ -57,14 +56,21 @@ func (lb *ELB) CreateIfNotExists() (bool, error) {
 		return false, microerror.Mask(clientNotInitializedError)
 	}
 
-	if err := lb.CreateOrFail(); err != nil {
-		if strings.Contains(err.Error(), awsclient.ELBConfigurationMismatch) {
-			return false, microerror.Mask(err)
-		}
-		if strings.Contains(err.Error(), awsclient.ELBAlreadyExists) {
-			return false, nil
-		}
+	lbDescription, err := lb.findExisting()
 
+	if err == nil {
+		// elb found, initialize dns and hoested zone id
+		lb.setDNSFields(*lbDescription)
+		return false, nil
+	}
+
+	if !strings.Contains(err.Error(), notFoundError.Error()) {
```

Comment:
We can't compare the errors directly?

Reply:
The actual error we are getting is `LoadBalancerNotFound: There is no ACTIVE Load Balancer named 'test-cluster-api'\n\tstatus code: 400, request id: aebfcf4d-bd87-11e7-89ff-f593dcba153f`, it includes the request id. Btw, instead of comparing with `notFoundError.Error()` we need to use here the `elb.ErrCodeAccessPointNotFoundException` constant from aws-sdk, it is the `LoadBalancerNotFound` part in the error string above. 

Do you think it would be better to use `strings.HasPrefix` instead of `strings.Contains`?

Reply:
I think either is fine, I was just wondering if there was a 'proper' matcher.